### PR TITLE
comply with libphonenumber's mainCountryForCode attribute when formatting numbers

### PR DESCRIPTION
**Pull request by fpbouchard**
see:
https://github.com/google/libphonenumber/blob/55716be400990f6542112323134cc1abc58abc2e/making-metadata-changes.md?plain=1#L34

Fixes: #74

### CODE REVIEW

**Comment by fpbouchard** [comment]
@cedvdb let me know what you think

**Comment by cedvdb** [comment]
great !

File: test/internal/_metadata_finder_test.dart [hunk 1]
```diff
@@ -35,6 +35,12 @@ void main() {
       }
     });
 
+    test('should get formats metadata for iso code going through references',
```
Comment (cedvdb) [inline test/internal/_metadata_finder_test.dart:38]: Could you add a more high level test in phone_number_test for the expected format behavior, for example for a canadian phone number ?

**Comment by cedvdb** [comment]
Sorry for the delay, I added a small comment.

File: test/internal/_metadata_finder_test.dart (hunk 1 above)
Comment (fpbouchard) [reply test/internal/_metadata_finder_test.dart:38]: Absolutely! Incoming.

**Pull request merged**